data_steward/cdr_cleaner/clean_cdr_engine.py

```diff
@@ -37,129 +41,225 @@ def add_console_logging(add_handler):
+        except (GoogleCloudError, TOError) as exp:
+            LOGGER.exception(
+                ce_consts.FAILURE_MESSAGE_TEMPLATE.render(
+                    project_id=client.project,
+                    **rule_info,
+                    **query_dict,
+                    exception=exp))
+            raise exp
+    return jobs
 
-            results = bq_utils.query(rule_query,
-                                     use_legacy_sql=legacy_sql,
-                                     destination_table_id=destination_table,
-                                     retry_count=retry,
-                                     write_disposition=disposition,
-                                     destination_dataset_id=destination_dataset,
-                                     batch=batch)
 
-        except (oauth2client.client.HttpAccessTokenRefreshError,
-                googleapiclient.errors.HttpError) as exp:
+def validate_params(clazz, **kwargs):
+    """
+    Filters kwargs based on the signature of the 'clazz'
 
-            LOGGER.exception(
-                format_failure_message(project_id=project,
-                                       statement=statement,
-                                       exception=exp))
-            failures += 1
-            continue
-
-        # wait for job to finish
-        query_job_id = results['jobReference']['jobId']
-        incomplete_jobs = bq_utils.wait_on_jobs([query_job_id])
-        if incomplete_jobs:
-            failures += 1
-            raise bq_utils.BigQueryJobWaitError(incomplete_jobs)
-
-        # check if the job is complete and an error has occurred
-        is_errored, error_message = bq_utils.job_status_errored(query_job_id)
-
-        if is_errored:
-            LOGGER.error(
-                format_failure_message(project_id=project,
-                                       statement=statement,
-                                       exception=error_message))
-            failures += 1
-        else:
-            if destination_table is not None:
-                updated_rows = results.get("totalRows")
-                if updated_rows is not None:
-                    LOGGER.info(
-                        f"Query returned {updated_rows} rows for {destination_dataset}.{destination_table}"
-                    )
-
-            successes += 1
-
-    if successes > 0:
-        LOGGER.info(
-            f"Successfully applied {successes} clean rules for {project}.{data_stage}"
-        )
-    else:
-        LOGGER.warning(
-            f"No clean rules successfully applied to {project}.{data_stage}")
+    :param clazz: Clean class or clean function to check the signature of
+    :param kwargs: optional arguments provided by the user
+    :return: filtered dictionary of kwargs
+    :raises ValueError: if a required param for 'clazz' is missing from kwargs
+    """
+    params = inspect.signature(clazz).parameters
+    mandatory_params = ['project_id', 'dataset_id']
```

Comment:
I recommend putting this in a consts file and renaming to something like `CLEAN_ENGINE_REQUIRED_PARAMS`